tests/Utility/InputValidatorTest.php

```diff
@@ -0,0 +1,396 @@
+					'is_numeric_array_key'    => false,
+					'is_stringable_object'    => false,
+					'is_accessible_as_array'  => false,
+					'is_iterable'             => false,
+					'is_curl_handle'          => false,
+				),
+			),
+			'empty array' => array(
+				'input'    => array(),
+				'expected' => array(
+					'is_string_or_stringable' => false,
+					'is_stringable_object'    => false,
+					'is_accessible_as_array'  => true,
+					'is_numeric_array_key'    => false,
+					'is_iterable'             => true,
+					'is_curl_handle'          => false,
+				),
+			),
+			'array with three items, no keys' => array(
+				'input'    => array(1, 2, 3),
+				'expected' => array(
+					'is_string_or_stringable' => false,
+					'is_numeric_array_key'    => false,
+					'is_stringable_object'    => false,
+					'is_accessible_as_array'  => true,
+					'is_iterable'             => true,
+					'is_curl_handle'          => false,
+				),
+			),
+			'array with two items, with keys' => array(
+				'input'    => array('a' => 1, 'b' => 2),
+				'expected' => array(
+					'is_string_or_stringable' => false,
+					'is_numeric_array_key'    => false,
+					'is_stringable_object'    => false,
+					'is_accessible_as_array'  => true,
+					'is_iterable'             => true,
+					'is_curl_handle'          => false,
+				),
+			),
+			'plain object' => array(
+				'input'    => new stdClass(),
+				'expected' => array(
+					'is_string_or_stringable' => false,
+					'is_numeric_array_key'    => false,
+					'is_stringable_object'    => false,
+					'is_accessible_as_array'  => false,
+					'is_iterable'             => false,
+					'is_curl_handle'          => false,
+				),
+			),
+			'object with __toString method' => array(
+				'input'    => new StringableObject('value'),
+				'expected' => array(
+					'is_string_or_stringable' => true,
+					'is_numeric_array_key'    => false,
+					'is_stringable_object'    => true,
+					'is_accessible_as_array'  => false,
+					'is_iterable'             => false,
+					'is_curl_handle'          => false,
+				),
+			),
+			'object implementing ArrayIterator' => array(
+				'input'    => new ArrayIterator(array(1, 2, 3)),
+				'expected' => array(
+					'is_string_or_stringable' => false,
+					'is_numeric_array_key'    => false,
+					'is_stringable_object'    => false,
+					'is_accessible_as_array'  => true,
+					'is_iterable'             => true,
+					'is_curl_handle'          => false,
+				),
+			),
+			'object implementing ArrayAccess' => array(
+				'input'    => new ArrayAccessibleObject(),
+				'expected' => array(
+					'is_string_or_stringable' => false,
+					'is_numeric_array_key'    => false,
+					'is_stringable_object'    => false,
+					'is_accessible_as_array'  => true,
+					'is_iterable'             => false,
+					'is_curl_handle'          => false,
+				),
+			),
+			'Curl handle (resource or object depending on PHP version)' => array(
+				'input'    => &self::$curl_handle,
```

Comment:
Why is this being passed as an explicit reference here? The way I understand it, it would be a reference for `CurlHandle` anyway, and not matter for a `resource`.

Reply:
Very good question. Copy/pasta somewhere down the line. Fixed now.